components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/service/WebSubEventSubscriberImpl.java

```diff
@@ -65,86 +64,78 @@ public class WebSubEventSubscriberImpl implements EventSubscriber {
     private static final Log log = LogFactory.getLog(WebSubEventSubscriberImpl.class);
 
     @Override
-    public String getName() {
+    public String getAssociatedAdaptor() {
 
         return WebSubHubAdapterConstants.WEB_SUB_HUB_ADAPTER_NAME;
     }
 
     @Override
-    public List<Subscription> subscribe(Webhook webhook, int tenantId) {
+    public List<Subscription> subscribe(WebhookSubscriptionRequest webhookSubscriptionRequest, String tenantDomain) {
 
         List<Subscription> subscriptions = new ArrayList<>();
-        String tenantDomain = IdentityTenantUtil.getTenantDomain(tenantId);
-        try {
-            for (Subscription eventSubscription : webhook.getEventsSubscribed()) {
-                try {
-                    makeSubscriptionAPICall(
-                            constructHubTopic(eventSubscription.getChannelUri(), webhook.getEventProfileVersion(),
-                                    tenantDomain), getWebSubBaseURL(), WebSubHubAdapterConstants.Http.SUBSCRIBE,
-                            webhook.getEndpoint(), webhook.getSecret());
-                    log.debug("WebSubHub subscription successful for channel: " + eventSubscription.getChannelUri() +
-                            " with endpoint: " + webhook.getEndpoint() + " in tenant: " + tenantDomain);
-
-                    Subscription subscription = Subscription.builder()
-                            .channelUri(eventSubscription.getChannelUri())
-                            .status(SubscriptionStatus.SUBSCRIPTION_ACCEPTED)
-                            .build();
-                    subscriptions.add(subscription);
-                } catch (WebSubAdapterException e) {
-                    log.debug("Error subscribing to channel" + eventSubscription.getChannelUri() + " with endpoint: " +
-                            webhook.getEndpoint() + " in tenant: " + tenantDomain + ". Error: " + e.getMessage(), e);
-                    Subscription subscription = Subscription.builder()
-                            .channelUri(eventSubscription.getChannelUri())
-                            .status(SubscriptionStatus.SUBSCRIPTION_ERROR)
-                            .build();
-                    subscriptions.add(subscription);
-                }
+        for (String channelToSubscribe : webhookSubscriptionRequest.getChannelsToSubscribe()) {
+            try {
+                makeSubscriptionAPICall(
+                        constructHubTopic(channelToSubscribe,
```

Comment:
shouldn't we pass event profile name here ?

Reply:
will fix this with the next pr, since need to change the topic management as well to pass the profile name.

Reply:
will streamline all the method signature to have the profile name